odata-json-format/4 Common Characteristics.md

```diff
@@ -423,17 +423,18 @@ and have the `null` value. In 4.01 payloads transient
 entities need not have the `id` control information, and 4.01
 clients MUST treat entities with neither `id` control
 information nor a full set of key properties as transient entities.
-
-The `id` control information MUST NOT appear for a
-collection. Its meaning in this context is reserved for future versions
-of this specification.
+Services MAY use a string value following the pattern `odata:transient:{some-generated-identifier-unique-within-the-response}` instead of `null`, for example if they  need to solve circular dependencies between transient entities by injecting an entity reference.
 
 Entities with `id` equal to `null` cannot be
-compared to other entities, reread, or updated. If
-[`metadata=minimal`](#metadataminimalodatametadataminimal)
+updated or compared to other entities.
+If [`metadata=minimal`](#metadataminimalodatametadataminimal)
 is specified and the `id` is not present in the entity, then
 the canonical URL MUST be used as the entity-id.
```

Comment:
This seems to contradict the preceding statement that 

> clients MUST treat entities with neither `id` control
> information nor a full set of key properties as transient entities.

Also it is not clear to me who must do this: the client or the service?